### PR TITLE
Mini change: allocate properly in aggregate

### PR DESCRIPTION
**Pull request by Bouncner**
Just a tiny thing I stumpled over. We do not reserve the null_vector when the column is not nullable. However, we always append to this vector.

### CODE REVIEW

File: src/lib/operators/aggregate_hash.cpp [hunk 1]
```diff
@@ -1033,7 +1033,7 @@ void AggregateHash::_write_groupby_output(RowIDPosList& pos_list) {
       auto values = pmr_vector<ColumnDataType>{};
       values.reserve(pos_list.size());
       auto null_values = pmr_vector<bool>{};
-      null_values.reserve(column_is_nullable ? pos_list.size() : 0);
```
[the remark this answers is not in the document]
Comment (Bouncner) [reply src/lib/operators/aggregate_hash.cpp:1036]: If I remember correctly, we benchmarked it. And the branch (even if it is probably always predicted right) was more expensive than writing the true/false. The best was to constexpr the loop, but I don't know if it is worth it here.

Or would you recommend something else?

[the remark this answers is not in the document]
Comment (dey4ss) [reply src/lib/operators/aggregate_hash.cpp:1036]: As discussed offline: I just wanted to make sure we don't set the null values vector when the column is not nullable. If the difference of having a (templated/flagged) function/lambda is negligible, maybe add a comment and we're fine.

**Comment by Bouncner** [comment]
Using the specialization of 5ca3a64 - b5a1f75 does not really change anything (Q18 is the query with the largest aggregate output). Would revert for simplicity.

<details>
<summary>Configuration Overview - click to expand</summary>

```diff
 +Configuration Overview----+------------------------------------------------+------------------------------------------------+
 | Parameter                | agg_master_sf100.json                          | agg_new_nulls_sf100.json                       |
 +--------------------------+------------------------------------------------+------------------------------------------------+
 |  GIT-HASH                | f7ad570cc30208c71593c0d7f609e34c0f60decb-dirty | 5ca3a648be5d6933d476b6fad464d57f6ad77ab7-dirty |
 |  benchmark_mode          | Ordered                                        | Ordered                                        |
 |  build_type              | release                                        | release                                        |
 |  chunk_indexes           | False                                          | False                                          |
 |  chunk_size              | 65535                                          | 65535                                          |
 |  clients                 | 1                                              | 1                                              |
 |  clustering              | None                                           | None                                           |
 |  compiler                | clang 14.0.0                                   | clang 14.0.0                                   |
 |  cores                   | 0                                              | 0                                              |
 |  data_preparation_cores  | 0                                              | 0                                              |
 |  date                    | 2023-03-14 00:29:08                            | 2023-03-13 23:26:22                            |
 |  encoding                | {'default': {'encoding': 'Dictionary'}}        | {'default': {'encoding': 'Dictionary'}}        |
 |  max_duration            | 600000000000                                   | 600000000000                                   |
 |  max_runs                | -1                                             | -1                                             |
 |  scale_factor            | 100.0                                          | 100.0                                          |
 |  time_unit               | ns                                             | ns                                             |
 |  use_prepared_statements | False                                          | False                                          |
 |  using_scheduler         | False                                          | False                                          |
 |  verify                  | False                                          | False                                          |
 |  warmup_duration         | 0                                              | 0                                              |
 +--------------------------+------------------------------------------------+------------------------------------------------+
```
</details>

```diff
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Item     || Latency (ms/iter)  | Change || Throughput (iter/s) | Change | p-value |
 |          ||      old |     new |        ||      old |      new |        |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | TPC-H 01 ||  58262.7 | 58841.2 |   +1%  ||     0.02 |     0.02 |   -1%  |  0.4377 |
 | TPC-H 18 ||  26034.3 | 25902.4 |   -1%  ||     0.04 |     0.04 |   +1%  |  0.5312 |
 +----------++----------+---------+--------++----------+----------+--------+---------+
 | Sum      ||  84297.0 | 84743.6 |   +1%  ||          |          |        |         |
 | Geomean  ||          |         |        ||          |          |   -0%  |         |
 +----------++----------+---------+--------++----------+----------+--------+---------+
```

**Comment by Bouncner** [comment]
The changes are part of #2556.

**Pull request closed without merging**